src/linalg/blas.c

```diff
@@ -82,6 +82,85 @@ igraph_error_t igraph_blas_dgemv(igraph_bool_t transpose, igraph_real_t alpha,
     return IGRAPH_SUCCESS;
 }
 
+/**
+ * \function igraph_blas_dgemm
+ * \brief Matrix-matrix multiplication using BLAS.
+ *
+ * This function is a somewhat more user-friendly interface to
+ * the \c dgemm function in BLAS. \c dgemm performs the operation
+ * y = alpha*a*b + beta*c, where a, b and c are matrices, of which a and b
+ * can be transposed.
+ *
+ * \param transpose_a whether to transpose the matrix \p a
+ * \param transpose_b whether to transpose the matrix \p b
+ * \param alpha       the constant \c alpha
+ * \param a           the matrix \c a
+ * \param b           the matrix \c b
+ * \param beta        the constant \c beta
+ * \param c           the matrix \c c. The result will also be stored here.
+ *                    If beta is zero, c wil be resized to fit the result.
+ *
+ * Time complexity: O(nmk) where matrix a is of size n x k, and matrix b is of
+ * size k x m.
+ *
+ * \return \c IGRAPH_EOVERFLOW if the matrix is too large for BLAS,
+ *         \c IGRAPH_EINVAL if the matrices have incompatible sizes,
+ *         \c IGRAPH_SUCCESS otherwise.
+ *
+ * \example examples/simple/blas_dgemm.c
+ */
+igraph_error_t igraph_blas_dgemm(igraph_bool_t transpose_a, igraph_bool_t transpose_b,
+        igraph_real_t alpha, const igraph_matrix_t* a, const igraph_matrix_t* b,
```

Comment:
Could we please write pointers as `type *var` instead of `type* var`, as in the rest of the codebase?

I always find `type* var` a bit misleading because `type* a, b` in fact declares only `a` as a pointer, not `b`. So the `*` attached to the var name, not the type name.

Reply:
I was wondering whether we should simply start using `clang-format`, set it up the way we like it, and then set our editors to call it upon saving. I started doing this in other C projects (in fact, I also added a pre-commit hook that pipes all changed source files through `clang-format`) and it works great and there is no need to take extra care when formatting code.